tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/ControllerActionInvoker_InvokeAction_Integration.cs

```diff
@@ -0,0 +1,80 @@
+// <copyright file="ControllerActionInvoker_InvokeAction_Integration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NETFRAMEWORK
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Web;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AspNet;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNet
+{
+    /// <summary>
+    /// System.Web.Mvc.Async.AsyncControllerActionInvoker.BeginInvokeAction calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = AssemblyName,
+        TypeName = "System.Web.Mvc.ControllerActionInvoker",
+        MethodName = "InvokeActionMethod",
+        ReturnTypeName = ActionResultTypeName,
+        ParameterTypeNames = new[] { ControllerContextTypeName, ActionDescriptorTypeName, DictionaryTypeName },
+        MinimumVersion = MinimumVersion,
+        MaximumVersion = MaximumVersion,
+        IntegrationName = IntegrationName)]
```

Comment:
If used only for appsec, it would be great to have a way to specify that here, in order not to rejit the method if appsec isn't enabled.

Reply:
yes, would another property on `InstrumentMethodAttribute` make sense, setting a condition like `OnlyAppSec`, and in `Instrumentation.Initialize` filtering the InstrumentationDefinitions depending on if AppSec is enabled? maybe @tonyredondo ?

Reply:
Fixed in PR https://github.com/DataDog/dd-trace-dotnet/pull/2580

Reply:
Now that it is merged we could add the filter here and in other instrumentations

Reply:
^ Do we still need to do this?

(this was marked resolve so I'm unresolving)

Reply:
Anna has done it but replied on the wrong comment. cf 

> added in https://github.com/DataDog/dd-trace-dotnet/commit/883ca333aad77f02baa945c32b3a349b40f16051

Reply:
Yes we were missing some commits from the old branch but now they're all back and we indeed use the filter